vitamins/belt.scad

```diff
@@ -64,8 +64,8 @@ module belt(type, points, gap = 0, gap_pos = undef, belt_colour = grey(20), toot
         if(gap)
             translate([gap_pos.x, gap_pos.y])
                 rotate(is_undef(gap_pos.z) ? 0 : gap_pos.z)
-                    translate([0, ph - thickness / 2])
-                        square([gap, thickness + eps], center = true);
+                    translate([0, is_list(gap) ? -th / 4 : ph - thickness / 2])
```

Comment:
Why tooth height / 4?

Reply:
That centers it on the gap.

Reply:
I don't see how. The belt is modelled relative to the pitch line, so adding the pitch height and subtracting half the thickness gets to the middle of the belt, so a rectangle slight thicker thicker than the belt just covers it.

How is the tooth height relevant and why divide it by 4?

Reply:
I'll look into it again.

Reply:
If gap.y is the offset from one end to the other that I think you need to add the belt thickness + eps to that to cover both. And the centre would be ph - thickness / 2 + gap.y / 2, I think.

Reply:
OK, I think it should be:
```
                    translate([0, (ph - thickness) / 2]);
```
ie, I think your original value of 
```
                    translate([0, ph - thickness / 2]);
```
is incorrect.

But in any case, the value does not need to change when a y value for the gap is set. ie the full code for `gap` is
```
    module gap()
        if(gap)
            translate([gap_pos.x, gap_pos.y])
                rotate(is_undef(gap_pos.z) ? 0 : gap_pos.z)
                    translate([0, (ph - thickness) / 2])
                        square(is_list(gap) ? [gap.x, gap.y]: [gap, thickness + eps], center = true);
```
But in any case, the difference is marginal, so I won't argue against it staying as `ph - thinkness / 2`.

Reply:
Why would you divide ph by 2? The pitch height is a magic number part way through the teeth measured from the tips of the teeth. Adding that gets to the tips and subtracting half the total thickness gets back to the centre, for a square centred on the belt. It looks spot on to me.

![image](https://user-images.githubusercontent.com/566149/99090494-a233fb00-25c6-11eb-8687-e0e71ef6cd65.png)

Reply:
OK, that makes sense. I had a misunderstanding of the pitch height.

Reply:
So how is the user specified y gap centred? Or do you pass in a length twice the actual gap?

Reply:
I think the correct y gap needs to be passed in because it also needs to be subtracted from the length.

Reply:
The y cap is a static property of the belts layout. It is the belt separation in the y axis plus the belt thickness. It certainly needs to be passed in as a parameter.

Reply:
So if you pass the actual y gap how is the square to remove it working without adding half the gap to centre it correctly?

And gap.y also needs to be subtracted from the length added to the BOM.

Reply:
Not sure what you mean about centering the gap, the call is:
```
square(is_list(gap) ? [gap.x, gap.y] : [gap, thickness + eps], center = true);
```
that is the center parameter is set to true.

As for the length of the belts, yes it needs to be subtracted, I forgot that.

Reply:
Yes I guess you just pass the centre of the gap in the ```gap_pos```. 

Does the y value you pass in need to be inflated by the belt width as it is from pitch line to pitch line? 

So I think you subtract gap.x + gap.y from the length but need a square gap.y + thickness + eps to remove it.

Reply:
Yes, the y value needs to be inflated by the thickness. In my code I pass in the the `ySeparation+2` just to be sure, but that would result in an incorrect belt length. So I think the parameter passed in should be just the separation (not inflated by the thickness) and the square should add the thickness, ie:
```
square(is_list(gap) ? [gap.x, gap.y + thickness + eps] : [gap, thickness + eps], center = true);
```

Reply:
Yes I think that is the way to go.